### ISSUE TITLE
Associated_token keywords can't be used except with `init`

### ISSUE THREAD

**Issue by Henry-E** [project-serum/anchor#814]
The `associated_token::` keywords require `init` to be in the account constraints. It would be nice if this wasn't required so that we could use it to verify that the submitted account is the correct associated token account.

Error message `init must be provided before token rustc`

e.g.
```rust
    #[account(mut, associated_token::mint = usdc_mint, associated_token::authority = user_authority)]
    pub user_usdc: Box<Account<'info, TokenAccount>>,
```

Created this originally in the solana program repo by accident 🤦

**Comment by acamill** [comment]
Also havn't tested, but what if the account is already initialized, a behaviour like "checkOrInit" would be neat

**Comment by armaniferrante** [comment]
Addressed by #843.
